library/std/src/net/ip.rs

```diff
@@ -958,18 +951,21 @@ impl PartialOrd<IpAddr> for Ipv4Addr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Ord for Ipv4Addr {
     fn cmp(&self, other: &Ipv4Addr) -> Ordering {
-        u32::from_be(self.inner.s_addr).cmp(&u32::from_be(other.inner.s_addr))
+        // We try to compare as native endian `u32` by using `u32::from_be`.
+        // In LE target, it swap bytes of BE inner to make it as LE `u32`.
+        // In BE target, it does nothing so we have BE `u32`.
+        u32::from_be(self.inner.to_be()).cmp(&u32::from_be(other.inner.to_be()))
```

Comment:
Doesn't this change behavior of the comparison function? Is that covered by a test?

Reply:
Yes, it is covered by tests.

Reply:
But then why do none of the tests change here? Or were tests broken on systems with some endianess before?

You didn't answer my question if this changes behavior.

Reply:
Sorry, it doesn't change old behaviour. I tried remove `u32::from_be` call and lots of tests fell.
The comparison was and is done with native endian u32.

Reply:
Ah I see.

Maybe change the text -- "try to compare" is strange, why "try"? We *do* compare in native endianess, right?